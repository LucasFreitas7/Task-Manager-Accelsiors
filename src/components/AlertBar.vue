<template>
    <v-row justify="center">
      <v-dialog
        v-model="dialog"
        persistent
        max-width="290"
        color="white"
      >
        <EditTask @close-dialog="closeDialog" :taskId="id" />
      </v-dialog>
    </v-row>
  </template>
  
  <script>
  import EditTask from '@/components/EditTask.vue'
    export default {
      name: 'AlertBar',
      components:{
        EditTask,
      },
      watch:{
        showDialog(newValue){
          this.dialog = newValue
        }
      },
      props:{
        id:{
          type: Number,
        },
        showDialog:{
          type: Boolean,
        },
      },
      data () {
      return {
        dialog: this.showDialog,
      }
    },
    methods:{
        closeDialog(){
            this.dialog = false,
            this.$emit('need-close')
        }
    },
    created(){
      console.log('Dialog on Alertbar =',  this.dialog)
    }
    }
  </script>
<style>

.v-dialog{
  background-color: white;
}
</style>
