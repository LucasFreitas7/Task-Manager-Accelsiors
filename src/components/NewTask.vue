<template>
    <v-container>
      <template>
        <div class="new-task">
            <v-select
              :items="items"
              label="Activity"
              v-model="task.activty"
            ></v-select>
            <v-text-field
                label="Message"
                hide-details="auto"
                v-model="task.comment"
            ></v-text-field>
            <v-text-field
            label="Duration *"
            hide-details="auto"
            type="number"
            v-model.number="task.duration"
            step="0.5"
            min="0.5"
            max="12"
            ></v-text-field>
            <div class="input-date">
              <v-col
              cols="12"
              sm="6"
              md="4"
              >
              <v-menu
                  ref="menu"
                  v-model="menu"
                  :close-on-content-click="false"
                  :return-value.sync="date"
                  transition="scale-transition"
                  offset-y
                  min-width="auto"
              >
              <template v-slot:activator="{ on, attrs }">
                  <v-text-field
                      v-model="date"
                      label="Date"
                      prepend-icon="mdi-calendar"
                      readonly
                      v-bind="attrs"
                      v-on="on"
                  ></v-text-field>
                  </template>
                  <v-date-picker
                  v-model="date"
                  no-title
                  scrollable
                  >
                  <v-spacer></v-spacer>
                  <v-btn
                      text
                      color="primary"
                      @click="menu = false"
                  >
                      Cancel
                  </v-btn>
                  <v-btn
                      text
                      color="primary"
                      @click="$refs.menu.save(date)"
                  >
                      OK
                  </v-btn>
                  </v-date-picker>
              </v-menu>
              </v-col>
          </div>
        <v-btn
            class="ma-2"
            color="green"
            :loading="loading"
            @click="newTask()"
            >
            New Task
        </v-btn>
        <AlertTask  v-if="warning != ''"   :message="warning"/>
        <AlertTask  v-if="warningSucess != ''" :message="warningSucess" :success="true"/>
        </div>
        </template>
    </v-container>
  </template>
  
  <script>
  import { mapGetters, mapActions } from "vuex";
  import AlertTask from "@/components/AlertTask.vue"
    export default {
      name: 'NewTask',
      components:{
        AlertTask
      },
      data: () => ({
        items: ['Cook', 'Videogame', 'Workout', 'Study', 'Walk'],
        date: (new Date(Date.now() - (new Date()).getTimezoneOffset() * 60000)).toISOString().substr(0, 10),
        menu: false,
        loading: false,
        warning: '',
        contId: 0,
        warningSucess: '',
        task: {},
      }),
      computed:{
        ...mapGetters(['getAll'])
      },
      methods:{
        ...mapActions(['addTask']),
        newTask(){
          this.warning = ''
          this.task.id = this.contId + 1
          this.contId++
          this.task.done = false
          this.task.date = this.date
          var exist = false
          this.getAll.map(e=> {
          if(e.date === this.task.date && e.activty ===  this.task.activty)
            exist = true
          })
          if(exist === true){
            this.warning = 'Already exist a activty on this day, please try again'
            setTimeout(() => {
              this.warning = ''
            },"3000")
            return
          }
          if((this.task.activty === ''  || !this.task.activty)  &&  (this.task.duration === '' || !this.task.duration)){
            this.warning = 'More then one inputs its missing'
            setTimeout(() => {
              this.warning = ''
            }, "3000")
            return
          }
          else if((this.task.activty === ''  || !this.task.activty)){
            this.warning = 'Activity is a required field'
            setTimeout(() => {
              this.warning = ''
            }, "3000")
            return
          }
          else if((this.task.duration === '' || !this.task.duration)){
            this.warning = 'Duration is a required field'
            setTimeout(() => {
              this.warning = ''
            }, "3000")
            return
          }
          else if(this.task.duration < 0.5 || this.task.duration > 12){
            this.warning = 'Duration needs to be in a range 0.5 to 12'
            setTimeout(() => {
              this.warning = ''
            }, "3000")
            return
          }
          this.addTask(this.task)
          this.task = {}
          this.warningSucess = 'Task created with success'
          setTimeout(() => {
            this.warningSucess = ''
          }, "2000")
          return
        }
  
      },
      created(){
        this.contId= this.getAll.length
      }
    }
  </script>
  
  <style>
  .col-md-4{
    padding: 0 !important;
  }

  .new-task{
    width: 40%;
  }

  @media (max-width: 600px) {
    .new-task{
      width: 90%;
    }
  }
  </style>
  