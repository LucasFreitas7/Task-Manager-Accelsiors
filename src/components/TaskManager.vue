<template>
  <v-container>
    <NewTask/>
    <TaskList @edit="fetchEditTask"/>
    <AlertBar  @need-close="changeEdit()" :showDialog="isToEdit" :id="editId"/>
  </v-container>
</template>

<script>
import NewTask  from '@/components/NewTask.vue'
import TaskList from '@/components/TasksList.vue'
import AlertBar from '@/components/AlertBar.vue'
  export default {
    name: 'TaskManager',
    data () {
      return {
        isToEdit: false,
        editId: null
      }
    },
    components:{
      NewTask,
      TaskList,
      AlertBar
    },
    methods:{
      fetchEditTask(value){
        this.isToEdit = true
        console.log('ITS to edit = ', this.isToEdit)
        this.editId = value
      },
      changeEdit(){
        console.log('Chegou aqui')
        this.isToEdit = false
        console.log('ITS to edit = ', this.isToEdit)
      }
    }
  }
</script>

<style>
.filter-activity{
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding: 10px;
}

.activity-text{
  padding-right: 10px;
}

.material-icons {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;  /* Preferred icon size */
  display: inline-block;
  line-height: 1;
  text-transform: none;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;

  /* Support for all WebKit browsers. */
  -webkit-font-smoothing: antialiased;
  /* Support for Safari and Chrome. */
  text-rendering: optimizeLegibility;

  /* Support for Firefox. */
  -moz-osx-font-smoothing: grayscale;

  /* Support for IE. */
  font-feature-settings: 'liga';
}

.panel-icons{
  padding-top: 10px;
}

.panel-buttons{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

</style>
