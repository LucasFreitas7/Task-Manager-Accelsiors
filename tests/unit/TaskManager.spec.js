import { shallowMount } from '@vue/test-utils'
import TaskManager from '@/components/TaskManager.vue'
import NewTask  from '@/components/NewTask.vue'
import TaskList from '@/components/TasksList.vue'
import AlertBar from '@/components/AlertBar.vue'
describe('TaskManager.vue', () => {

  it('Snapshot test', () => {
    const wrapper = shallowMount(TaskManager)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Component NewTask must be exist', () => {
    const wrapper = shallowMount(TaskManager)
    const newTask = wrapper.findComponent(NewTask)
    expect(newTask.exists()).toBe(true)
  })
  it('Component TaskList must be exist', () => {
    const wrapper = shallowMount(TaskManager)
    const taskList = wrapper.findComponent(TaskList)
    expect(taskList.exists()).toBe(true)
  })
  it('Component AlertBar must be exist', () => {
    const wrapper = shallowMount(TaskManager)
    const alertBar = wrapper.findComponent(AlertBar)
    expect(alertBar.exists()).toBe(true)
  })
})
