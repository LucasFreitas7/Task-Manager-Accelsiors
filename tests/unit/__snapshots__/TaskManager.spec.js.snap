// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskManager.vue Snapshot test 1`] = `
<v-container>
  <newtask-stub></newtask-stub>
  <tasklist-stub></tasklist-stub>
  <alertbar-stub></alertbar-stub>
</v-container>
`;
